### PR TITLE
JP-3312 Support Update from RHEL7 to RHEL9

### LINKED ISSUE #7753: Support Update from RHEL7 to RHEL9
_Issue [JP-3312](https://jira.stsci.edu/browse/JP-3312) was created on JIRA by [Ian Broussard](https://jira.stsci.edu/secure/ViewProfile.jspa?name=ibroussard):_

Per JSOCINT-712, we need to develop a plan to move from Redhat Linux 7 to 8.  This is also a Cross Division Engineering Initiative being tracked under CDEI-71.  *The end-of-life for REL7 is June 2024.*

We need the component leads for CAL to document what work will be needed to upgrade any use of REL7 or older, and what build it would be feasible to get that work done in.  We are targeting Q1 of 2024 to have this work completed, but each component can let us know if Q1 2024 or earlier would work for them.

### PR DESCRIPTION
**Pull request by penaguerrero**
Resolves [JP-3312](https://jira.stsci.edu/browse/JP-3312)

Closes #7753 

This PR addresses the failing reg test test_miri_image3_catalog. The tolerance was not being propagated in the function call due to a missing parameter, when added the test passes.

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [ ] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [ ] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -26,6 +26,12 @@ outlier_detection
 - Fixed failures due to a missing ``wcs.array_shape`` attribute when the
   ``outlier_detection`` step was run standalone using e.g. ``strun`` [#8645]
 
+regtest
```
Comment (melanieclarke) [inline CHANGES.rst:29]: I don't think this needs a change log entry, since there's no impact to users.

**Review by melanieclarke — COMMENTED** [review]
Seems okay to me!

**Comment by braingram** [comment]
As a proxy for RHEL9 here's a link to the github actions regression tests run with this PR:
https://github.com/spacetelescope/RegressionTests/actions/runs/9959697888
as noted in the ticket, the differences seen for RHEL9 look identical to those seen with ubuntu 20.04 (the OS used for github actions).

**Review by melanieclarke — APPROVED** [review]
Looking at the regression tests, it looks like there were some CRDS related issues in the psfalign tests, but the remainder all look like the expected diffs listed in JP-3312.  The miri_image3_catalog test passed, so I think we can go ahead and merge this one.

**Pull request merged**